### PR TITLE
Added react land date before and after

### PR DESCRIPTION
**Pull request by tomkingston**
## Description of change

Add "Estimated Land date" date field filters to the React Investments collection list

## Test instructions

Navigate to /investments/projects/ and see the two date field filters - when applied, the list should be filtered.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/client/components/DateField/index.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    font-weight: ${FONT_WEIGHTS.regular};
+  }
+`
+
+const DateField = ({ ...props }) => {
+  const { error, onChange, name, label, legend, hint, initialValue } = props
+  return (
+    <StyledFieldWrapper {...{ name, label, legend, hint, error }}>
+      <StyledInputWrapper error={error}>
+        {error && <ErrorText>{error}</ErrorText>}
```
Comment (ian-leggett) [inline src/client/components/DateField/index.jsx:37]: We can remove this as we are not handling any errors in the filters.

File: src/client/components/DateField/index.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  BORDER_WIDTH_FORM_ELEMENT_ERROR,
+  FONT_WEIGHTS,
+  SPACING,
+} from '@govuk-react/constants'
+import ErrorText from '@govuk-react/error-text'
+import Input from '@govuk-react/input'
+
+import FieldWrapper from '../Form/elements/FieldWrapper'
+
+const StyledInputWrapper = styled('div')`
```
Comment (ian-leggett) [inline src/client/components/DateField/index.jsx:16]: We can remove this as we are not handling any errors in the filters.

File: src/client/components/DateField/index.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+`
+
+const StyledFieldWrapper = styled(FieldWrapper)`
+  label {
+    font-weight: ${FONT_WEIGHTS.regular};
+  }
+`
+
+const DateField = ({ ...props }) => {
+  const { error, onChange, name, label, legend, hint, initialValue } = props
```
Comment (ian-leggett) [inline src/client/components/DateField/index.jsx:33]: We can deconstruct this in the function parenthesis, we tend to do this on other components. If we do this we can also just return the JSX implicitly.

File: src/client/components/DateField/index.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  onChange: PropTypes.func,
+  name: PropTypes.string.isRequired,
+  label: PropTypes.node,
+  legend: PropTypes.node,
+  hint: PropTypes.string,
+  initialValue: PropTypes.string,
+}
+
+DateField.defaultProps = {
+  onChange: () => {},
```
Comment (ian-leggett) [inline src/client/components/DateField/index.jsx:61]: Although we have used defaultProps in other places I think now we are trying to set the default value in the function parenthesis. Example `const func({onchange = ()=> {}}) => // do something` Let me know what you think?

File: src/client/components/DateField/index.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    <StyledFieldWrapper {...{ name, label, legend, hint, error }}>
+      <StyledInputWrapper error={error}>
+        {error && <ErrorText>{error}</ErrorText>}
+        <Input
+          id={name}
+          name={name}
+          error={error}
+          defaultValue={initialValue}
+          type="date"
+          onChange={(e) => onChange(e)}
```
Comment (ian-leggett) [inline src/client/components/DateField/index.jsx:44]: This line could change to `onChange={onChange}`

**Review by ian-leggett — COMMENTED** [review]
This is looking great! Just a couple of small comments let me know what you think, also I think we need to add the indicator chips to this PR (sorry I should have said and this should have been in the story card). Also I noticed you named the stories dir "no-stories", we have been naming them "__stories__" could this be renamed? Any reason for "no-stories"? Thanks 👍

File: src/client/components/RoutedDateField/index.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    {({ location, history }) => {
+      const qsParams = qs.parse(location.search.slice(1))
+      return (
+        <DateField
+          {...props}
+          onChange={(e) => {
+            history.push({
+              search: qs.stringify({
+                ...qsParams,
+                [qsParamName]: e.target.value,
```
Comment (ian-leggett) [inline src/client/components/RoutedDateField/index.jsx:19]: Just under this line we need to also reset the paging to 1, so this needs to be ```
            history.push({
              search: qs.stringify({
                ...qsParams,
                [qsParamName]: e.target.value,
                page: 1,
              }),
            })
```
This is missing in the sort too so I better look at this!

Comment (tomkingston) [reply src/client/components/RoutedDateField/index.jsx:19]: Good point - didn't think about that!

**Review by rachaelcodes — APPROVED** [review]

**Review by ian-leggett — APPROVED** [review]

**Pull request merged**